rfcs/convergence/apperance-migration.md

```diff
@@ -0,0 +1,106 @@
+
+👍 Pros:
+
+- Is relatively safe
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Composed component apperance prop wouldn't match our Fluent V9 documentation
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+- Composition approach is currently marked as unstable in v9
+
+### Wrap the library components on application side
+
+Partners could a new component wrapper that will wrap input components and modify the appearance prop on the application side.
+
+👍 Pros:
+
+- Is safe and stable
+
+👎 Cons:
+
+- Creating new composed component for each input `Dropdown`, `Input`, `TextArea` and `DatePicker`
+- Wrapper component apperance wouldn't match our Fluent V9 documentation
+- Would work in iframes only within the TMP repository
+- Does not work if repos have dependencies of another project with Fluent V9 input components (Nova)
+
+### Global css selector
+
+Targeting all input selectors from a partner app and change the color with global css.
+
+Example: https://codesandbox.io/s/rfc-inputs-ew821q?file=/app.tsx
+
+👍 Pros:
+
+- Relatively easy and fast to do
+- Scalable
+- Would work for dependencies such as Nova (but depending where we inject the CSS rule and if they have their own `FluentProvider`)
+
+👎 Cons:
+
+- Difficult to validate
+
+### New token alias to theme
+
+Adding a new alias color token (let's say `colorInputBackground`) and use it for all inputs.
+
+👍 Pros:
+
+- Easy to do and would work for all inputs
+
+👎 Cons:
+
+- Negative impact on performance by increasing variables (as read here: [fluentui/theme-shared-colors.md at d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351 · microsoft/fluentui (github.com)](https://github.com/microsoft/fluentui/blob/d5d510bf1ffcc1a4ed2067e9eb009c84e7beb351/rfcs/react-components/convergence/theme-shared-colors.md))
+- Divergence themes from the original
+
+Another option is to add a possibility to override the background using a CSS variable:
+We can use `backgroundColor: var(--inputBackgroundOverride, ${tokens.colorNeutralBackground1})` without setting the `--inputBackgroundOverride` anywhere. Then an application can set that variable if it needs to override the background.
+
+👍 no additional tokens unless needed
+
+👎 new concept
+👎 one-off just for the Input background
+
+### Unify design
+
+Discuss with designers to unify V0 and V9 design, setting the appearance to filled-dark by default.
+
+👍 Props:
+
+- Will make migration easier for partners who already uses V0
+- Would work with iframes (if all the teams have the unified design)
+
+👎 Cons:
+
+- Inherits design from old V0 package that does not meet our needs/goals
+- According to the design team, this is currently a no-go
```

Comment:
```suggestion
- According to the design team, this is currently a no-go

### Use React context

Another option is to use React Context to override `appearance` defaults in `FluentProvider`.

#### Example

```tsx
function App() {
  return (
    <FluentProvider appearance="filled-darker">
      <Input /> {/* has "filled-darker" */}
      <Input appearance="underline" /> {/* has "underline" */}
    </FluentProvider>
  )
}
```


```

Let's list one more option to explore 🐱